core/secret.go

```diff
@@ -12,7 +12,7 @@ func (secret *Secret) CreateSecret(secretString string) bool {
 		println(err.Error())
 		return false
 	}
-	secret.data = data[len(data) - 26 : 16]
+	secret.data = data[0 : 16]
```

Comment:
Thanks, my bad, this is fair.  Guess you can just data[:16]

Btw for me this code works fine (I've tested on Linux and MacOS machines). Just wondering why it doesn't work for you?

Reply:
I was getting error, because secret isn't long enough. I made fix like month ago without any knowledge about Go.

I have linux too, but i wasn't able to run binary. I got error with length of secret.